src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializerNamespaces.cs

```diff
@@ -86,51 +84,64 @@ public XmlQualifiedName[] ToArray()
         /// </devdoc>
         public int Count
         {
-            get { return Namespaces.Count; }
+            get { return (_namespaces == null) ? 0 : _namespaces.Count; }
         }
 
-        internal ArrayList? NamespaceList
+        internal IEnumerable<XmlQualifiedName> Namespaces => NamespacesInternal.Values;
+
+        private Dictionary<string, XmlQualifiedName> NamespacesInternal
         {
             get
             {
-                if (_namespaces == null || _namespaces.Count == 0)
-                    return null;
-                ArrayList namespaceList = new ArrayList();
-                foreach (string key in Namespaces.Keys)
-                {
-                    namespaceList.Add(new XmlQualifiedName(key, (string?)Namespaces[key]));
-                }
-                return namespaceList;
+                if (_namespaces == null)
+                    _namespaces = new Dictionary<string, XmlQualifiedName>();
+                return _namespaces;
             }
         }
 
-        [AllowNull]
-        internal Dictionary<string, string?> Namespaces
+        internal ArrayList? NamespaceList
```

Comment:
Why **ArrayList**?

Reply:
The diff in the review makes it look like these two are related. They're not. 'internal ArrayList? NamespaceList' was in the original code on line 92. I'm not a fan of converting to yet another array-ish format via this property, but there was one place in the XML code that was exposing this through a public property. So it had to stay. :/